test/langtools/tools/javac/patterns/Guards.java

```diff
@@ -84,6 +92,31 @@ String testBooleanSwitchExpression(Object o) {
         }
     }
 
+    String typeGuardIfTrueSwitchStatement(Object o) {
+        Object o2 = "";
+        switch (o) {
+            case Integer i && i == 0 && i < 1 && o2 instanceof String s: o = s + String.valueOf(i); return "true";
```

Comment:
could you please add tests with `||` in the condition?

Reply:
Thanks, Vicente. I've added tests with `||` in https://github.com/openjdk/jdk17/pull/41/commits/d9e902b02b8a9916954fe4d2b665d4eaee3fb352 . One thing to note is that the tests are a bit different when writing something like: `case String s && s.isEmpty() || o2 instanceof Number n ->` `n` is not available here, as there is a path on which `o2` is not a `Number`.